### ISSUE TITLE
Problem with derivatives when use_ws_distance = True

### ISSUE THREAD

**Issue by giovannipizzi** [wannier-developers/wannier90#48]
There is some problem when calculating band derivatives with the new interpolation method use_ws_distance = True

Probably, we are still using the old 'r' in the computation of the gradient?

See also results of geninterp for the two interpolation methods below (last three columns are the velocity)

[new.txt](https://github.com/wannier-developers/wannier90/files/476156/new.txt)
[old.txt](https://github.com/wannier-developers/wannier90/files/476157/old.txt)

**Comment by sponce24** [comment]
It seems that there might also be a problem with use_ws_distance = False in parallel. 
I will open an issue about it.

**Comment by jryates** [comment]
On 19 Sep 2016, at 19:09, Samuel Ponc� <notifications@github.com> wrote:

> It seems that there might also be a problem with use_ws_distance = False in parallel. 
> I will open an issue about it.

It is not broadcast at the start of a parallel run. It needs to be added to list of broadcast variables in postw90_common
 
 Jonathan




-- 
Department of Materials, University of Oxford, Parks Road, Oxford, OX1 3PH, UK
tel: +44 (0)1865 612797                http://users.ox.ac.uk/~oums0549/

**Comment by giovannipizzi** [comment]
I think the original issue was solved in #53. About the new issue, I just created #65: Samuel, can you confirm us if this solves the issue? (And additional question: is this anyway related to #64?)

**Comment by sponce24** [comment]
It seems that it does not solve #64 

http://129.67.86.21:8010/builders/WAN-farmer_intel13/builds/24/steps/test_wannier/logs/stdio

**Comment by giovannipizzi** [comment]
Ok, in the meantime I mark this issue as closed and we'll look deeper into #64
